azurerm/internal/services/network/parse/network_connection_monitor.go

```diff
@@ -0,0 +1,40 @@
+package parse
+
+import (
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type NetworkConnectionMonitorId struct {
+	ResourceGroup string
+	WatcherId     string
+	WatcherName   string
+	Name          string
+}
+
+func NetworkConnectionMonitorID(input string) (*NetworkConnectionMonitorId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, err
+	}
+
+	connectionMonitor := NetworkConnectionMonitorId{
+		ResourceGroup: id.ResourceGroup,
+		WatcherId:     strings.Split(input, "/connectionMonitors/")[0],
```

Comment:
Can we insist on using the `id.Popsegment` to do the id parsing?

Reply:
updated